lib/ruby_lsp/requests/support/prefix_tree.rb

```diff
@@ -0,0 +1,81 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    module Support
+      class PrefixTree
+        extend T::Sig
+
+        sig { returns(Node) }
+        attr_reader :root
```

Comment:
Do we need to expose `root` here? This seems like an internal implementation detail and people should be reaching for `insert` and `search` instead.

Reply:
Removed!